src/lib/sql/sql_translator.cpp

```diff
@@ -466,7 +497,27 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_table_origin(const hsq
 
   switch (hsql_table_ref.type) {
     case hsql::kTableName: {
-      if (StorageManager::get().has_table(hsql_table_ref.name)) {
+      // WITH descriptions or sub-queries are treated as though they were inline views or tables
+      // They mask existing tables or views with the same name.
+      if (_with_descriptions.count(hsql_table_ref.name)) {
+        const auto iter = _with_descriptions.find(hsql_table_ref.name);
+        AssertInput(iter != _with_descriptions.end(), "No such WITH query named '" + hsql_table_ref.name + "'");
```

Comment:
How could that ever trigger if `_with_descriptions.count(hsql_table_ref.name) > 0`, as the if below ensures?

Reply:
Correct 🙈